src/Azure/Orleans.Streaming.EventHubs/Providers/Streams/EventHub/EventHubStreamOptions.cs

```diff
@@ -30,6 +34,27 @@ public class EventHubStreamOptions : RecoverableStreamOptions
         /// </summary>
         public bool StartFromNow { get; set; } = DEFAULT_START_FROM_NOW;
         public const bool DEFAULT_START_FROM_NOW = true;
+    }
+
+    public class EventHubStreamCacheOptions
```

Comment:
I understand these are the options for the eventhub cache, but these options are not really eventhub specific, and they seem to be more tightly related to a purge strategy which can be used with many other stream providers.  I'm wondering if something like DataAgeCachePurgeOptions might be more appropriate and allow the components using these settings to be used across a wider range of providers.

Reply:
yes that's what I originally did. But then I think not all cache has purge logic. So how do I prevent users to configure that purge option with stream providers which doesn't support purging.

I'm still unsure which route to take. Just want to bring my thinking here

Reply:
Only exposing the configuration of this purge strategy in providers that support it (which is currently event hub atm, correct?) should be sufficiently limiting for now, imo.

The distinction here is the difference between what is being configured (a purge strategy) and the provider which utilizes it.  In theory, this purge strategy can be used by many other stream providers.  The fact that it is currently only being used by the event hub stream provider does not make it an event hub stream provider purge strategy.

Reply:
I may have misunderstood your question.. For eh stream provider, i'd suggest a ConfigureCachePurgeStrategy (or something of the sort) rather than the current ConfigureCache.  Sure, we'll want a ConfigureCache at some point to allow customization, but for configuring the defaults, since all we can configure is the purge strategy, that's all I'd suggest we expose at this time.

Reply:
but we are not  just configuring purge here, we are also configure cache pressure monitors. 

The cache purge settings can be reused in MemoryStreamProvider. So does this mean we have two methods, one is configure purge, which both eh and memory stream provider has it, and another is cache monitor?

Reply:
Good point.  Yes, I think options should be associated with the components they configure, so options related to backpressure monitoring should be associated with the backpressure monitor.
We may need to change the default behavior here to better fit the pattern as well as future expectations.  Having no back pressure monitoring by default, expecting users to opt into the monitoring they expect, seems appropriate at this time.  So maybe an AddAveragingCachePressureMonitoring call with the related options?

Reply:
I'd love to go in that direction, like make all those component injectable. But with this overrlap with EventHUbQueueCacheFactory.AddCachePressureMonitor, ,and how eh streaming currently is. It is rather big change, with some source of truth problems.  I think we can do it in another pr after rc2 ?